packages/components/src/components/post-rating/post-rating.scss

```diff
@@ -0,0 +1,98 @@
+@use '@swisspost/design-system-styles/core' as post;
+
+:host {
+  display: inline-block;
+}
+
+.rating {
+  margin: 0 (-(post.$size-micro));
+  display: flex;
+  flex-wrap: wrap;
+  box-sizing: border-box;
+  overflow: hidden;
+  & :focus {
+    outline: none;
+  }
+  &[aria-readonly='true'] .star-container {
+    cursor: default;
+  }
+}
+
+.rating:focus-visible {
+  outline-offset: post.$input-focus-outline-thickness;
+  outline: post.$input-focus-outline-thickness solid post.$input-focus-color;
+  position: relative;
+  z-index: 1;
+}
+
+.star-container {
+  position: relative;
+  display: block;
+  width: post.$lead-font-size;
+  height: post.$lead-font-size;
+  padding: 0 post.$size-micro 0 post.$size-micro;
+  cursor: pointer;
```

Comment:
You could use the approch to declair css variables here for fill and border, use them in `.fill` and `.border` classes and then only switch the css variables, when a modifier class (`.active-star`, `.hovered-star`, etc.) is appended to the `.star-container` element.

Also nest `.fill` and `.border` rules inside of the `.star-container` to make clear, they need to be a child of it.

```scss
.star-container {
  ...
  --post-rating-star-fill-color: #{post.$white};
  --post-rating-star-border-color: #{post.$black};
  
  .fill {
    position: absolute;
    color: var(--post-rating-star-fill-color);
  }
  
  .border {
    position: absolute;
    ...
    color: var(--post-rating-star-border-color);
  }
  
  &.active-star {
    --post-rating-star-fill-color: #{post.$yellow};
    --post-rating-star-border-color: #{post.$black};
  }
  
  ...
}
```